### DISCUSSION TITLE
Issue setting up AIO behind nginx-proxy reverse proxy

### DISCUSSION THREAD

**Discussion by firesurfer** [nextcloud/all-in-one#553] [Questions] [answered]
Hi I am trying to setup nextcloud AIO behind an nginx reverse proxy which also runs in a docker container (https://github.com/nginx-proxy/nginx-proxy).

I manage to get to the AIO containers page which asks me for the domain I am hosting the nextcloud on. There I enter my domain and always get the message: `Domain does not point to this server or reverse proxy not configured correctly.` 

I can definitely verify that the domain is right (I am actually accessing the page via that domain). I also verified that from inside the nextcloud master container (by starting a shell inside) the domain is reachable on port 443. 

My docker compose file is:

```
version: '3.6'

networks:
  nextcloud:
services:
  nginx-proxy:
    image: 'nginxproxy/nginx-proxy'
    ports: 
       - "80:80"
       - "443:443"
    volumes: 
       - ./nginx_proxy/config:/etc/nginx/conf.d
       - ./nginx_proxy/vhost:/etc/nginx/vhost.d
       - ./nginx_proxy/html:/usr/share/nginx/html
       - ./nginx_proxy/dhparam:/etc/nginx/dhparam
       - ./nginx_proxy/certs:/etc/nginx/certs:ro
       - /var/run/docker.sock:/tmp/docker.sock:ro
    networks:
       - default
       - nextcloud
       
  nextcloud:
    image: 'nextcloud/all-in-one:latest'
    container_name: 'nextcloud-aio-mastercontainer'
    environment: 
       VIRTUAL_HOST: nextcloud.myinternaldomain.de
       VIRTUAL_PORT: 8080
       CERT_NAME: wildcard.myinternaldomain.de
       VIRTUAL_PROTO: https
       APACHE_PORT: 11000
    volumes:
       #note: the nextcloud container wants a global volume and does not accept a local mounted folder....
       - nextcloud_aio_mastercontainer:/mnt/docker-aio-config
       - /var/run/docker.sock:/var/run/docker.sock:ro
    ports:
       - "3478:3478"
       - "8080:8080"
    expose:
       - 8080

    networks:
       - nextcloud

 
volumes:
  nextcloud_aio_mastercontainer:
    name: nextcloud_aio_mastercontainer
```

Note I explicitly do not want to specify the `networking host` option as I want seperated network stacks!
The reverse proxy setup works btw for other services like gitlab without any problems.

**Comment by szaimen** [comment]
I suppose this goes into the same direction like https://github.com/nextcloud/all-in-one/discussions/543. You need to specify the config in some conf file of your nginx proxy and not using environmental variables or labels for the docker container itself to make it work. Also network host option is required for the reverse proxy if running inside a docker container to make it work...

If someone figured this out I'll gladly add it to the documentation then.

**Reply by firesurfer** [reply]
>  You need to specify the config in some conf file of your nginx proxy

Can you elaborate this. What config file are you refering to ?

> Also network host option is required for the reverse proxy if running inside a docker container to make it work...

Also regarding this point some more information would be great. As I said there are many other services like gitlab, youtrack etc which work fine with this kind of configuration.

**Reply by szaimen** [reply]
yes, it might work for gitlab and other containers but Nextcloud AIO is special :)
See https://github.com/nextcloud/all-in-one#how-does-it-work

**Reply by szaimen** [reply]
so basically it spawns the so called `domaincheck` container with the chosen APACHE_PORT upon the startup of the all-in-one container which when you enter the domain in the interface the reverse proxy needs to connect to. After that was successful it will spawn the apache container which will use the configured APACHE_PORT. that is why using `localhost:$APACHE_PORT` and the network host option is recommended way to configure the reverse proxy

**Reply by firesurfer** [reply]
Thanks for your answers and effort. I decided that the normal nextcloud container fits my usage scenario better

**Reply by szaimen** [reply]
Okay

**Comment by szaimen** [comment, marked as answer]
Unfortunately nginx-proxy will likely never be supported. See the updated reverse proxy documentation why: https://github.com/nextcloud/all-in-one/blob/main/reverse-proxy.md

If you really want to use AIO, I'll recommend to switch to caddy. It is simply amazing.
Of course understandable if that is not possible for you.

Apart from that, there is this idea: https://github.com/nextcloud/all-in-one/issues/557
Pull requests are very welcome :)
